### PR TITLE
Course list optimizations

### PR DESCRIPTION
**Pull request by bjrne**
such as using course tab for launch and adding a working segmented control for all/my courses

### CODE REVIEW

File: xikolo-common/ViewControllers/AbstractCourseListViewController.swift [hunk 1]
```diff
@@ -28,17 +28,21 @@ class AbstractCourseListViewController : UICollectionViewController {
...
-                resultsController = CoreDataHelper.createResultsController(request, sectionNameKeyPath: "is_enrolled_section")
+        case .EnrolledOnly:
+            request = CourseHelper.getMyCoursesRequest()
+            resultsController = CoreDataHelper.createResultsController(request, sectionNameKeyPath: nil)
+        case .All:
+            request = CourseHelper.getAllCoursesRequest()
+            resultsController = CoreDataHelper.createResultsController(request, sectionNameKeyPath: nil)
+        case .BothSectioned:
+            request = CourseHelper.getSectionedRequest()
+            resultsController = CoreDataHelper.createResultsController(request, sectionNameKeyPath: "is_enrolled_section")
```
Comment (invliD) [inline xikolo-common/ViewControllers/AbstractCourseListViewController.swift]: Remove these whitespace-only changes.

File: xikolo-ios/ViewControllers/Courses/CourseListViewController.swift [hunk 2]
```diff
@@ -10,8 +10,25 @@ import UIKit
...
+            showMyCoursesOnly(false)
+        case 1:
+            if UserProfileHelper.isLoggedIn() {
+                showMyCoursesOnly(true)
+            } else {
+                sender.selectedSegmentIndex = 0
+                performSegueWithIdentifier("ShowLoginForMyCourses", sender: sender) // maybe switch to My Courses after succesful login?
+            }
+        default:
+            showMyCoursesOnly(true)
```
Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseListViewController.swift]: Is this default case necessary? It should only ever be 0 or 1, right?

Comment (bjrne) [inline xikolo-ios/ViewControllers/Courses/CourseListViewController.swift]: In our case, it should. But in theory, segmented controls can consist of more options, so Swift wants a default clause here. Is there a way to tell it we only have those two values here?

Comment (invliD) [inline xikolo-ios/ViewControllers/Courses/CourseListViewController.swift]: Then let's just do `default: break` here.

**Pull request merged**